client/rest/src/main/java/org/elasticsearch/client/ResponseException.java

```diff
@@ -39,18 +39,21 @@ public ResponseException(Response response) throws IOException {
     }
 
     private static String buildMessage(Response response) throws IOException {
-        String message = response.getRequestLine().getMethod() + " " + response.getHost() + response.getRequestLine().getUri()
-                + ": " + response.getStatusLine().toString();
+        final StringBuilder sb = new StringBuilder("method [");
+        sb.append(response.getRequestLine().getMethod());
+        sb.append("], host [").append(response.getHost());
+        sb.append("], URI [").append(response.getRequestLine().getUri());
+        sb.append("], status line [").append(response.getStatusLine().toString()).append(']');
```

Comment:
I appreciate the use of `StringBuilder`, but I think we can do something simpler and clearer here:

```java
String.format(
    Locale.ROOT,
    "method [%s], host [%s], URI [%s], status line [%s]\n%s",
    . 
    .
    .);
```

I find this so much easier to see what the end result is going to be.